node/src/config.rs

```diff
@@ -0,0 +1,150 @@
+use miden_node_block_producer::config::BlockProducerConfig;
+use miden_node_rpc::config::RpcConfig;
+use miden_node_store::config::StoreConfig;
+use miden_node_utils::config::Config;
+use serde::{Deserialize, Serialize};
+
+pub const CONFIG_FILENAME: &str = "miden-node.toml";
+
+// Top-level config
+// ================================================================================================
+
+/// Node top-level configuration.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize, Default)]
+pub struct NodeTopLevelConfig {
+    pub block_producer: BlockProducerConfig,
+    pub rpc: RpcConfig,
+    pub store: StoreConfig,
+}
+
+impl Config for NodeTopLevelConfig {
+    const CONFIG_FILENAME: &'static str = CONFIG_FILENAME;
+}
+
+#[cfg(test)]
+mod tests {
+    use figment::Jail;
+    use miden_node_block_producer::config::BlockProducerConfig;
+    use miden_node_rpc::config::RpcConfig;
+    use miden_node_store::config::StoreConfig;
+    use miden_node_utils::{config::HostPort, Config};
+
+    use super::{NodeTopLevelConfig, CONFIG_FILENAME};
+
+    #[test]
+    fn test_node_config() {
+        Jail::expect_with(|jail| {
+            jail.create_file(
+                CONFIG_FILENAME,
+                r#"
+                    [block_producer]
+                    store_endpoint = "http://store:8000"
+
+                    [block_producer.host_port]
+                    host = "127.0.0.1"
+                    port = 8080
+
+                    [rpc]
+                    store_endpoint = "http://store:8000"
+                    block_producer_endpoint = "http://block_producer:8001"
+                    host_port = { host = "127.0.0.1",  port = 8080 }
+
+                    [store]
+                    sqlite = "local.sqlite3"
+
+                    [store.host_port]
+                    host = "127.0.0.1"
+                    port = 8080
```

Comment:
Some naming comments:

- I'd probably use `endpoint` instead of `host_port`.
- I'd probably use `url` instead of `endpoint` for non-host endpoints.

So, this could look something like this:
```Toml
[block_producer]
endpoint = { host = "127.0.0.1", port = 8080 }
store_url = "http://store:8000"
     
[rpc]
endpoint = { host = "127.0.0.1", port = 8080 }
store_url = "http://store:8000"
block_producer_url = "http://block_producer:8001"

[store]
endpoint = { host = "127.0.0.1", port = 8080 }
sqlite = "local.sqlite3"
```